src/Eshopworld.WorkerProcess/Telemetry/LeaseExceptionEvent.cs

```diff
@@ -1,18 +1,23 @@
-﻿using Eshopworld.Core;
+﻿using System;
 
 namespace EShopworld.WorkerProcess.Telemetry
 {
     /// <summary>
     /// An event that is raised when an exception occurs when an exception occurs during the leasing process
     /// </summary>
-    internal class LeaseExceptionEvent : ExceptionEvent
+    public class LeaseExceptionEvent : BaseLeaseEvent
     {
         /// <summary>
         ///     Create instance of <see cref="LeaseExceptionEvent" />
         /// </summary>
-        /// <param name="exception"></param>
-        public LeaseExceptionEvent(System.Exception exception) : base(exception)
+        /// <param name="instanceId"></param>
+        /// <param name="workerType"></param>
+        /// <param name="priority"></param>
+        /// <param name="reason"></param>
+        public LeaseExceptionEvent(Guid instanceId, string workerType, int priority, string reason) : base(instanceId,workerType,priority,reason)
```

Comment:
Are we not catching the exception details in the event?

Reply:
Is that still 'exception' event?

Reply:
I noticed that nothing was being logged in AI with the previous event, so i used the BaseLeaseEvent as base class. This also to clearly identify what insttanceId is breaking. should we possibly log the stacktrace instead of Message, or even better having an additional field Details with the stacktrace?

Reply:
I would add the exception here but you may need to inherit from exception event (it should log it?). But I see you re-throw the exception so not the end of the world

Reply:
correct. I tried to keep the existing behaviour as much as possible.